modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/CachingEvaluator.scala

```diff
@@ -0,0 +1,110 @@
+package com.snowplowanalytics.snowplow.enrich.common.enrichments.registry
+
+import cats.Monad
+import cats.effect.Clock
+import cats.implicits._
+import com.snowplowanalytics.lrumap.{CreateLruMap, LruMap}
+import com.snowplowanalytics.snowplow.enrich.common.enrichments.registry.CachingEvaluator.{Cache, CachedItem, GetResult, Value}
+
+import java.util.concurrent.TimeUnit
+
+final class CachingEvaluator[F[_]: Clock: Monad, K, V](
+  cache: Cache[F, K, V],
+  config: CachingEvaluator.Config
+) {
```

Comment:
The pattern we use elsewhere in Snowplow code is that classes do not have implicit type classes like `Clock` or `Monad`.  And instead we put the type class on the methods:

```
final class CachingEvaluator[F[_], K, V](....) {
  def evaluateForKey(....)(implicit M: Monad[F], C: Clock[F]): F[Either[Throwable, V]] =
    ???
}
```

To be completely honest, the people who cared most about this pattern have now left Snowplow!!  So in a way we are free to ignore that rule and do what we want :)  And unfortunately I don't know the full details of why they liked that pattern.

Would it be problematic to use the old pattern here too?

Reply:
It wouldn't be problematic at all. If it's the pattern, let's be consistent! I will change it.